### PR TITLE
feat: resource nav mesh

### PR DESCRIPTION
**Pull request by MarcGonzalezMoratona**

### CODE REVIEW

File: Source/DataModels/Resources/ResourceNavMesh.h [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+	bool GetKeepInterResults() const;
+
+	void SetKeepInterResults(bool keepInterResults);
+
+	unsigned char GetNavMeshDrawFlags() const;
+	
+	void SetNavMeshDrawFlags(unsigned char navMeshDrawFlags);
+
+public:
+	enum DrawMode
```
Comment (gerard-queralt) [inline Source/DataModels/Resources/ResourceNavMesh.h:73]: Prefer use of scoped enums
```suggestion
	enum class DrawMode
```

File: Source/FileSystem/Importers/NavMeshImporter.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "DataModels/Resources/ResourceNavMesh.h"
+#include "FileSystem/Importers/Importer.h"
+
+class NavMeshImporter : public Importer<ResourceNavMesh>
+{
+public:
+	NavMeshImporter(){};
+	~NavMeshImporter(){};
```
Comment (gerard-queralt) [inline Source/FileSystem/Importers/NavMeshImporter.h:9]: Missing `override` identifier
```suggestion
	~NavMeshImporter() override {};
```

File: Source/FileSystem/Importers/NavMeshImporter.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+	void Import(const char* filePath, std::shared_ptr<ResourceNavMesh> resource) override;
+	void Load(const char* fileBuffer, std::shared_ptr<ResourceNavMesh> resource) override;
+
+protected:
+	void Save(const std::shared_ptr<ResourceNavMesh>& resource, char*& fileBuffer, unsigned int& size) override;
```
Comment (gerard-queralt) [inline Source/FileSystem/Importers/NavMeshImporter.h:15]: Was about to comment that I'd change the output parameter to be the return type of the method, but I've noticed that this is an inherited method. We can probably change this in a different MR.

Comment (MarcGonzalezMoratona) [reply Source/FileSystem/Importers/NavMeshImporter.h:15]: we can create a draft PR to change this everywhere. Please create a task and I will take it when I have some free time

File: Source/FileSystem/Importers/NavMeshImporter.h (hunk 2 above)
Comment (MarcGonzalezMoratona) [reply Source/FileSystem/Importers/NavMeshImporter.h:9]: I added it at the beginning and I don't know why I removed it. Nice catch!

File: Source/DataModels/Resources/ResourceNavMesh.h [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+	int detailSampleDist = 6;
+	int detailSampleMaxError = 1;
+
+	bool keepInterResults = false;
+
+	int tileSize = 48;
+	int maxTiles = 0;
+	int maxPolysPerTile = 0;
+
+	DrawMode drawMode = DRAWMODE_NAVMESH;
```
Comment (gerard-queralt) [inline Source/DataModels/Resources/ResourceNavMesh.h:117]: Looks like the pipeline is failing, try scoping the enum value, that should fix it.
```suggestion
	DrawMode drawMode = DrawMode::DRAWMODE_NAVMESH;
```

File: Source/DataModels/Resources/ResourceNavMesh.h [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+	//class dtNavMeshQuery* navQuery = nullptr;
+	//class dtCrowd* crowd = nullptr;
+
+	//class dtTileCache* tileCache = nullptr;
+
+	//struct LinearAllocator* talloc = nullptr;
+	//struct FastLZCompressor* tcomp = nullptr;
+	//struct MeshProcess* tmproc = nullptr;
+
+	unsigned char navMeshDrawFlags = 0;
```
Comment (gerard-queralt) [inline Source/DataModels/Resources/ResourceNavMesh.h:131]: Is there a more appropiate type for these flags? Just so whoever reads or writes them has an easy way of knowing what each byte represents.

Comment (MarcGonzalezMoratona) [reply Source/DataModels/Resources/ResourceNavMesh.h:131]: we will address the final types after physics finishes with the nav mesh implementation. For the moment this is a placeholder resource

**Review by gerard-queralt — APPROVED** [review]

File: Source/DataModels/Resources/ResourceNavMesh.h [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+	int detailSampleDist = 6;
+	int detailSampleMaxError = 1;
+
+	bool keepInterResults = false;
+
+	int tileSize = 48;
+	int maxTiles = 0;
+	int maxPolysPerTile = 0;
+
+	DrawMode drawMode = DrawMode::DRAWMODE_NAVMESH;
```
Comment (AuxPort66) [inline Source/DataModels/Resources/ResourceNavMesh.h:117]: We didn't save this DrawMode Enum, probably goes inside the meta. We can check it when we have more information about the functionality.

**Review by AuxPort66 — COMMENTED** [review]
I see the importer and resource part correct, but the next time don't mix random refactors because navMesh should go to physics-general first, but the rest of this PR should go to develop to avoid future conflicts.

We need a placeholder method to create this NavMesh resource (Like a create resource button on filebrowser). I can help you with that so we can test this properly meanwhile Physics put the new library.

**Pull request closed without merging**